packages/template-block/src/components/PreviewImage.tsx

```diff
@@ -0,0 +1,101 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { AppBridgeBlock, Template, useBlockAssets, useEditorState } from '@frontify/app-bridge';
+import { PreviewType, Settings, previewDisplayValues, previewImageAnchoringValues } from '../types';
+import { IconArrowSync, IconSpeechBubbleQuote20, merge } from '@frontify/fondue';
+import { BlockItemWrapper } from '@frontify/guideline-blocks-settings';
+import { EditAltTextFlyout } from '@frontify/guideline-blocks-shared';
+import { useEffect, useState } from 'react';
+
+export type PreviewImageProps = {
+    appBridge: AppBridgeBlock;
+    blockSettings: Settings;
+    template: Template | null;
+    updateBlockSettings: (newSettings: Partial<Settings>) => Promise<void>;
+    onOpenTemplateChooser: () => void;
+};
+
+export const PreviewImage = ({
+    appBridge,
+    blockSettings,
+    template,
+    updateBlockSettings,
+    onOpenTemplateChooser,
+}: PreviewImageProps) => {
+    const isEditing = useEditorState(appBridge);
+    const { blockAssets } = useBlockAssets(appBridge);
+    const { preview, previewImageAnchoring, previewDisplay } = blockSettings;
+    const previewSrc =
+        preview === PreviewType.Custom && blockAssets.previewCustom !== undefined
+            ? blockAssets.previewCustom[0].previewUrl
+            : template?.previewUrl;
+    const [currentPreviewSrc, setCurrentPreviewSrc] = useState(previewSrc);
+    const [showAltTextMenu, setShowAltTextMenu] = useState(false);
+    const [localAltText, setLocalAltText] = useState<string | undefined>(blockSettings.altText);
+    const [isHovered, setIsHovered] = useState(false);
+
+    useEffect(() => {
+        setCurrentPreviewSrc(previewSrc);
+    }, [preview, template, blockAssets.previewCustom]);
+
+    return (
+        <>
+            <img
+                data-test-id="template-block-preview-img"
+                src={currentPreviewSrc}
+                className={merge(['tw-relative tw-w-full tw-h-full', previewDisplayValues[previewDisplay]])}
+                style={{
+                    objectPosition: previewImageAnchoring
+                        ? previewImageAnchoringValues[previewImageAnchoring]
+                        : 'center',
+                }}
+                width={
+                    preview === PreviewType.Custom && blockAssets.previewCustom
+                        ? blockAssets.previewCustom[0].width
+                        : 'auto'
+                }
+                height={
+                    preview === PreviewType.Custom && blockAssets.previewCustom
+                        ? blockAssets.previewCustom[0].height
+                        : 'auto'
+                }
+                alt={blockSettings.altText || undefined}
+                onMouseEnter={() => setIsHovered(true)}
+                onMouseLeave={() => setIsHovered(false)}
+                onBlur={() => setIsHovered(false)}
```

Comment:
is `onFocus={()=>setIsHovered(true}` missing? Otherwise if this cannot be focused, than maybe onBlur is not necessary?

Reply:
You're right. `onBlur` here is not needed. Cleaned up.